refactor: Major refactoring of functions to improve readability, efficiency and follow standard practices.

holtskinner opened the pull request:
- Shouldn't impact any user-facing functions

BEGIN_COMMIT_OVERRIDE

fix: Internal refactoring to improve efficiency and readability. No external-facing changes.

END_COMMIT_OVERRIDE

holtskinner committed:
refactor: Reorganize hocr functions

- Use more jinja templating instead of hardcoding strings
- Simplified bounding box function
- Changed parameter name for `_get_hocr_bounding_box` to `page_dimension` for more clarity.

holtskinner committed:
samples: Added sample for convert to hocr

holtskinner committed:
refactor: Reordering of classes in page.py

holtskinner committed:
refactor: Re-added refactoring to remove extra `get_*()` methods in page.py

- Added in https://github.com/googleapis/python-documentai-toolbox/pull/110 Lost in Merge

holtskinner committed:
fix: Moved `templates` directory into package.
- Required for template to work in installed library

holtskinner committed:
chore: Ran isort and black

holtskinner committed:
chore: Ran no-implicit-optional

holtskinner committed:
refactor: Refactored document.py
- improve readability, follow python conventions, and improve efficiency
- Also, fixed a previously unknown bug where `Document.search_pages()` returned inaccurate results because it only searched paragraph.text, not page.text

holtskinner committed:
refactor: Refactor gcs_utilities for readability/pythonic style

holtskinner committed:
refactor: Refactor page.py to improve efficiency, readability and follow python conventions

renovate-bot committed:
chore(deps): update dependency google-cloud-documentai to v2.16.1 (#138)

holtskinner committed:
refactor: Rename `Entity.documentai_entity` to `Entity.documentai_object` to match the page.py file

holtskinner committed:
refactor: Move bounding box extraction to `docai_utilities.py`

holtskinner committed:
refactor: Major Refactoring of converter_helpers.py to simplify/organize functions, reduce complexity, and increase readability

holtskinner committed:
fix: Fixed refactor of export_images in document.py

holtskinner committed:
refactor: Cleanup of blocks.py using `getattr()`

holtskinner committed:
refactor: Refactoring of bbox_conversion.py to improve readability and efficiency

holtskinner committed:
Merge branch 'main' into linting

holtskinner committed:
fix: Change _get_files() to send full gcs uri to _get_bytes()
- Also reduce wait_time in tests

holtskinner committed:
fix: Change _get_files() to send full gcs uri to _get_bytes()
- Also reduce wait_time in tests

holtskinner committed:
refactor: Move `converter_helpers.py` functions into `converter.py`

- `converter.py` only had one external facing function that called an internal function with the same parameters.
- Not sure if there was a specific reason for this setup, can be undone if needed.

holtskinner committed:
refactor: Move `converter_helpers.py` functions into `converter.py`

- `converter.py` only had one external facing function that called an internal function with the same parameters.
- Not sure if there was a specific reason for this setup, can be undone if needed.

holtskinner committed:
Merge branch 'linting' of https://github.com/googleapis/python-documentai-toolbox into linting

holtskinner committed:
Merge branch 'main' into linting

holtskinner committed:
chore: Reran black formatting after merge conflict

holtskinner committed:
refactor: Minor refactoring of test_bbox_conversion.py to improve readability

holtskinner committed:
refactor: Changed blocks.py to block.py for consistency.

- Changed how `Block` is initialized.
- Changed `load_blocks_from_schema` into a `@classmethod` to simplify imports.

holtskinner committed:
fix: Added Missing type annotations to `document.py`

holtskinner committed:
Merge branch 'main' into linting

holtskinner committed:
fix: Add new filename for block.py into test_bbox_conversion.py

gcf-owl-bot[bot] committed:
🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

holtskinner committed:
fix: Fix failing tests for Block class. Changed all fields to have types

holtskinner committed:
Merge branch 'linting' of https://github.com/googleapis/python-documentai-toolbox into linting

holtskinner committed:
fix: Changed `converter._get_bytes` to return a Tuple

holtskinner committed:
Merge branch 'main' into linting

holtskinner committed:
Merge branch 'main' into linting

dizcology commented on google/cloud/documentai_toolbox/converters/config/bbox_conversion.py:
```diff
@@ -130,27 +133,19 @@ def _convert_bbox_units(
+        "pxl": 1,
+        "inch": 96,
+        "cm": 37.795,
+    }
```
Can we put these in constants?

holtskinner replied:
Do you mean make the whole variable a constant? Or constants for the strings/Numbers?

dizcology commented on google/cloud/documentai_toolbox/converters/config/bbox_conversion.py:
```diff
@@ -109,7 +108,11 @@ def _convert_to_pixels(x: float, conversion_rate: float) -> float:
 def _convert_bbox_units(
-    coordinate, input_bbox_units, width=None, height=None, multiplier=1
+    coordinate: float,
+    input_bbox_units: str,
```
Can `input_bbox_units` be an enum instead of `str`?

holtskinner replied:
I agree that would make more sense. However, since this value is provided by the user in a JSON file, I wonder if there is a straightforward way to make that conversion? (Besides changing the config value to be an integer) Or maybe if it would be better to use constants?

dizcology commented on google/cloud/documentai_toolbox/converters/config/block.py:
```diff
@@ -0,0 +1,239 @@
+
+        blocks: List[Block] = []
+        ens = _get_target_object(objects, entities)
+        for i in ens:
```
The logic of this for loop and its nested if-else clauses is difficult to understand.  I get the impression that many of them could be simplified if a block `b`'s attributes have reasonable defaults.  For instance, can't

```
if bounding_width:
                b.bounding_width = _get_target_object(b.bounding_box, bounding_width)
```

be simplified to just
```
b.bounding_width = _get_target_object(b.bounding_box, bounding_width)
```
?

(Please ignore and file an internal clean up issue if this file was simply renamed.)

holtskinner replied:
This was a refactoring that I did as well. So I'm good to make some changes as needed.

dizcology commented on google/cloud/documentai_toolbox/converters/converter.py:
```diff
@@ -15,9 +15,368 @@
+FILES_TO_IGNORE = [".DS_Store"]
+
+
+def _get_base_ocr(
```
Are these simply moved from converter_helpers?  Let me know if these additions need a more detailed review.

holtskinner replied:
These were refactored as well, and everything was moved into this file.

You can see just the refactoring work in this commit https://github.com/googleapis/python-documentai-toolbox/pull/139/commits/fb517fba8154e8e8da0553b8c8b3467a39061786

holtskinner committed:
chore: Addressed Code Review Comments

- Removed FILES_TO_IGNORE
- Simplification of logic in `_get_multiplier` `convert_bbox_to_docproto_bbox`
- Addressed other lint errors
- Adjusted function names to indicate not protected members.

holtskinner committed:
fix: Remove extra reference to metadata_blob

holtskinner committed:
fix: Change expected test output and remove references to `geometry`

holtskinner committed:
Merge branch 'main' into linting
